src/use-localstorage.ts

```diff
@@ -48,19 +58,16 @@ export function useLocalStorage<TValue = string>(key: string, initialValue?: TVa
         }
       }
     }
-  }, [updateLocalState]);
-
+  };
 
   useEffect(() => {
-    // The custom storage event allows us to update our component 
+    // The custom storage event allows us to update our component
     // when a change occurs in localStorage outside of our component
-    window.addEventListener(
-      LocalStorageChanged.eventName,
-      (e: any) => onLocalStorageChange(e as LocalStorageChanged<TValue>)
-    );
+    const listener = (e: Event) => onLocalStorageChange(e as LocalStorageChanged<TValue>);
```

Comment:
A stable pointer is set inside the `useEffect`. This renders the `useCallback` wrapping `onLocalStorageChange` function redundant.